app/src/main/java/ca/llamabagel/transpo/ui/search/SearchViewModel.kt

```diff
@@ -4,32 +4,49 @@
 
 package ca.llamabagel.transpo.ui.search
 
+import android.util.Log
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import ca.llamabagel.transpo.data.SearchRepository
 import ca.llamabagel.transpo.ui.search.viewholders.SearchResult
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 enum class KeyboardState {
     OPEN, CLOSED
 }
 
-class SearchViewModel @Inject constructor(private val searchRepository: SearchRepository) : ViewModel() {
+@FlowPreview
+@ExperimentalCoroutinesApi
+class SearchViewModel @Inject constructor(
+    private val searchUseCase: GetSearchResultsUseCase,
+    private val updateQueryUseCase: UpdateQueryUseCase
+) : ViewModel() {
 
     private val _keyboardState = MutableLiveData<KeyboardState>().apply { value = KeyboardState.OPEN }
     val keyboardState: LiveData<KeyboardState> = _keyboardState
 
     private val _searchResults = MutableLiveData<List<SearchResult>>().apply { value = emptyList() }
     val searchResults: LiveData<List<SearchResult>> = _searchResults
 
-    fun notifyClosed() {
-        _keyboardState.value = KeyboardState.CLOSED
+    fun startListeningToSearchResults() = viewModelScope.launch {
+        searchUseCase.invoke().collect { _searchResults.postValue(it) }
```

Comment:
```suggestion
        getSearchResults().collect { _searchResults.postValue(it) }
```

Reply:
again just checking tests... most of this stuff isn't really ready for review